lib/mp4/transmuxer.js

```diff
@@ -887,17 +887,17 @@ CoalesceStream.prototype.flush = function(flushSource) {
 
   initSegment = mp4.initSegment(this.pendingTracks);
 
-  // Create a new typed array large enough to hold the init
-  // segment and all tracks
-  event.data = new Uint8Array(initSegment.byteLength +
-                              this.pendingBytes);
+  // Create a new typed array to hold the init segment
+  event.initSegment = new Uint8Array(initSegment.byteLength);
```

Comment:
Originally I attempted to "pass down" whether we wanted to create the init segment by setting the flag whenever a discontinuity is encountered and `setBaseMediaDecodeTime` is called. However, we also need to add the init segment on audio configuration changes, which the transmuxer has no knowledge of. Since the transmuxer has been sending the init segment data along with every segment to the source buffers anyway, we are not really adding more overhead by continuing to send it over. I feel it simplifies things to let the transmuxer just handle transmuxing and leave the decision to media-sources